File: backend/app/schemas/document_approval.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentApprovalBase(BaseModel):
    approval_type: str = "approval"
    status: str = "pending"
    comment: str | None = None


class DocumentApprovalCreate(DocumentApprovalBase):
    document_revision_id: uuid.UUID
    tenant_id: uuid.UUID
    approver_user_id: uuid.UUID


class DocumentApprovalAction(BaseModel):
    comment: str | None = None


class DocumentApprovalResponse(DocumentApprovalBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    document_revision_id: uuid.UUID
    tenant_id: uuid.UUID
    approver_user_id: uuid.UUID
    acted_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class DocumentApprovalListResponse(BaseModel):
    items: list[DocumentApprovalResponse]
    total: int

File: frontend/src/types/documentApproval.ts
export type DocumentApprovalRecord = {
  id: string;
  document_revision_id: string;
  tenant_id: string;
  approver_user_id: string;
  approval_type: string;
  status: string;
  comment: string | null;
  acted_at: string | null;
  created_at: string;
  updated_at: string;
};

export type DocumentApprovalListResponse = {
  items: DocumentApprovalRecord[];
  total: number;
};